include/crm/common/xml_compat.h

```diff
@@ -71,6 +71,10 @@ crm_element_name(const xmlNode *xml)
     return (xml == NULL)? NULL : (const char *) xml->name;
 }
 
+//! \deprecated Do not use (will be removed in a future release)
```

Comment:
Another option would be to keep the public `xml_create_patchset()` function and throw an error on `format == 1` unless `PCMK_ALLOW_DEPRECATED`.

The same goes for `xml_apply_patchset()`.

We could deprecate both of them later, after we already have libpacemaker equivalents that don't accept a format (or that don't support v1).

Reply:
I suppose to future-proof against a v3, it might be worthwhile for at least the top functions in the call chain to insist on format=2. If there's a v3 it should never be sent to a node that can't understand it, but that would be a failsafe in case it happened anyway

Reply:
> future-proof

I think the latest updates accomplish this. There are a few functions that still don't check the format, because the patchsets are always generated by the local Pacemaker instance and thus will be the latest version:
```
based_callbacks.c:send_peer_reply()
controld_te_callbacks:te_update_diff()
crm_mon.c:crm_diff_update()
crm_diff.c:generate_patch()
pacemaker-fenced.c:update_fencing_topology()
```